Source/aweXpect.Reflection/ThatTypeHas.OrHave.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+using aweXpect.Core;
+using aweXpect.Core.Constraints;
+using aweXpect.Reflection.Helpers;
+using aweXpect.Results;
+
+namespace aweXpect.Reflection;
+
+public static partial class ThatType
+{
+	/// <summary>
+	///     Allows an alternative attribute of type <typeparamref name="TAttribute" />.
+	/// </summary>
+	/// <remarks>
+	///     The optional parameter <paramref name="inherit" /> (default value <see langword="true" /> specifies, if
+	///     the attribute can be inherited from a base type.
+	/// </remarks>
+	public static AndOrResult<Type?, IThat<Type?>> OrHave<TAttribute>(
```

Comment:
You should only be able to call `OrHave<TAttribute>()` after you called `Have<TAttribute>()`, so this should not be an extensio method on `AndOrResult<Type>`, but you should change the return value of `Have<TAttribute>()` to a class that has the `OrHave<TAttribute>()` methods.